content/chapters/compute/lab/support/sum-array/c/sum_array_processes.c

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/mman.h>
+#include <sys/wait.h>
+
+#include "generate_random_array.h"
+
+#define ARR_LEN 100000000
+
+#define min(a, b)                       \
+	({                              \
+		__typeof__(a) _a = (a); \
+		__typeof__(b) _b = (b); \
+		_a < _b ? _a : _b;      \
+	})
+
+static void calculateArrayPartSum(int *array, size_t start, size_t end,
+				  long *results, size_t tid)
+{
+	long sumArray = 0;
+
+	for (size_t i = start; i < end; ++i)
+		sumArray += array[i];
+
+	results[tid] = sumArray;
+}
+
+static inline void *createSharedResultsArray(size_t numProcesses)
+{
+	// This buffer will be shared by the process that creates it and all its
+	// children due to the `MAP_SHARED` argument
+	return mmap(NULL, numProcesses * sizeof(long),
+		    PROT_READ | PROT_WRITE, MAP_SHARED | MAP_ANON, -1, 0);
+}
+
+int main(int argc, char *argv[])
+{
+	int numProcesses = atoi(argv[1]);
+	struct timeval start, stop;
+	long time;
+	pid_t children[numProcesses];
+	int ret;
+	long *results = (long *)createSharedResultsArray(numProcesses);
+	long finalResult = 0;
+	int *array;
+
+	if (argc < 2)
+	{
+		fprintf(stderr, "Usage: %s <num_threads>\n", argv[0]);
+		exit(1);
+	}
+
+	array = malloc(sizeof(*array) * ARR_LEN);
+	if (!array)
+	{
+		fprintf(stderr, "Error when allocating array: %s\n", strerror(errno));
+		exit(1);
+	}
```

Comment:
The [generic makefiles](https://github.com/open-education-hub/operating-systems/tree/master/content/common/makefile) already import the [`DIE()` macro](https://github.com/open-education-hub/operating-systems/blob/f51e235230370c8a4855ec612772b6745c57101a/content/common/utils/utils.h#L23-L30), which you can use here.